src/nat/middleware/dynamic/dynamic_function_middleware.py

```
@@ -0,0 +1,959 @@
+    # ==================== FunctionMiddleware Interface Implementation ====================
+
+    @property
+    def enabled(self) -> bool:
+        """Whether this middleware should execute.
+
+        Returns config.enabled value. Framework checks this before invoking
+        any middleware methods.
+        """
+        return self._config.enabled
+
+    async def pre_invoke(self, context: PreInvokeContext) -> PreInvokeContext | None:
```

Ruff (0.14.8)

Unused method argument: `context`

(ARG002)